libs/langchain_v1/langchain/agents/middleware/model_fallback.py

```diff
@@ -93,6 +93,7 @@ def wrap_model_call(
         # Try fallback models
         for fallback_model in self.models:
             request.model = fallback_model
+            request.model_settings = {}
```

Comment:
Could you do this using `override`?  request = request.override(model=fallback_model)